src/test/java/uk/gov/hmcts/reform/api/filters/SensitiveHeadersRequestTraceFilterTest.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.hmcts.reform.api.filters;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.entry;
+
+public class SensitiveHeadersRequestTraceFilterTest {
+
+    @Test
+    public void should_filter_out_sensitive_headers_ignoring_case() {
+
+        // given
+        Map<String, Object> headers = new HashMap<>();
+
+        headers.put("OK Header 1", "xxx");
```

Comment:
For readability, it would be worth extracting the creation of the header map out of this test method, so that we would have three lines here - one for non-sensitive headers, one for sensitive headers and one for all headers.

Reply:
There's only one test here, not need to scatter the code around multiple functions imho.
Would you be happy with adding comments on what's being done in given sections of test?